source/common/stats/tag_extractor_impl.cc

```diff
@@ -114,13 +114,27 @@ bool TagExtractorStdRegexImpl::extractTag(TagExtractionContext& context, std::ve
                                                      regex_) &&
       match.size() > 1) {
     // remove_subexpr is the first submatch. It represents the portion of the string to be removed.
-    const auto& remove_subexpr = match[1];
+    uint32_t idx;
```

Comment:
rather than doing all of this, WDYT of changing the Istio regexes to only match one or two things?

You could also add something like:
```
if (match.size() > 2) {
  ENVOY_LOG_MISC(error, "tag extraction matched too many regexes");
  return false;
}
```